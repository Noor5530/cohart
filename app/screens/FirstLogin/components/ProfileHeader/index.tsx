import React from "react";
import { Image, Text, View, useWindowDimensions } from "react-native";

import styles from "./style";
interface Props {
  heading: string;
  description: string;
  renderDescription?: React.ReactElement;
}
const Profile: React.FC<Props> = (props: Props) => {
  const { width } = useWindowDimensions();
  const { heading, description } = props;

  return (
    <View style={[styles.container, { width }]}>
      <Image
        style={styles.logo}
        source={require("assets/profileHeaderIcon.png")}
      />
      <View>
        <Text style={styles.header}>{heading}</Text>
        {props.renderDescription ? (
          props.renderDescription
        ) : (
          <Text style={styles.description}>{description}</Text>
        )}
      </View>
    </View>
  );
};

export default Profile;
