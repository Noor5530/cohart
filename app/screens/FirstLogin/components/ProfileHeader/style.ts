import {StyleSheet} from "react-native"

import { appFonts } from 'components/text';
import { primary } from 'config/colors';
import { iconColor } from 'config/colors';
import { heightPercentageToDP as hp, widthPercentageToDP as wp } from "react-native-responsive-screen";

const styles = StyleSheet.create({
    container: {
      height: hp('30'),
      backgroundColor: primary,
      justifyContent: 'flex-end',
      paddingBottom: 30,
    },
    logo: {
      width: wp('22'),
      height: hp('18'),
      position: 'absolute',
      top: 0,
      right: 0
    },
    header: {
      paddingLeft: 38,
      fontFamily: appFonts.InterBold,
      height: 19,
      fontStyle: 'normal',
      fontWeight: 'bold',
      fontSize: hp('2'),
      lineHeight: 19, 
      textTransform: 'uppercase',
      color: iconColor,
    },
    description: {
      fontFamily: appFonts.InterRegular,
      paddingLeft: 38,
      fontStyle: 'normal',
      fontWeight: 'normal',
      fontSize: hp('1.6'),
      textTransform: 'uppercase', 
      color: iconColor,
    },
  });
  
  export default styles