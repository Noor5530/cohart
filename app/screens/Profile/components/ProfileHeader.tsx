import { appFonts } from 'components/text';
import { primary } from 'config/colors';
import React from 'react';
import { Image, StyleSheet, Text, View } from 'react-native';

import { iconColor } from '../../../config/colors';

interface Props {
  heading: string;
  description: string;
}
const Profile: React.FC<Props> = (props: Props) => {
  const { heading, description } = props;

  return (
    <View style={styles.container}>
      <Image
        style={styles.logo}
        source={require('assets/profileHeaderIcon.png')}
      />
      <View>
        <Text style={styles.header}>{heading}</Text>
        <Text style={styles.description}>{description}</Text>
      </View>
    </View>
  );
};

export default Profile;

const styles = StyleSheet.create({
  container: {
    width: '100%',
    height: 216.05,
    backgroundColor: primary,
    justifyContent: 'flex-end',
    paddingBottom: 30,
  },
  logo: {
    width: 88,
    height: 154,
    position: 'absolute',
    bottom: -10,
    right: 0,
  },
  header: {
    paddingLeft: 38,
    fontFamily: appFonts.InterBold,
    height: 19,
    fontStyle: 'normal',
    fontWeight: 'bold',
    fontSize: 15,
    lineHeight: 19,

    textTransform: 'uppercase',

    color: iconColor,
  },
  description: {
    fontFamily: appFonts.InterRegular,
    paddingLeft: 38,
    fontStyle: 'normal',
    fontWeight: 'normal',
    fontSize: 8.28337,
    paddingTop: 12,
    lineHeight: 12 /* identical to box height */,

    textTransform: 'uppercase',

    color: iconColor,
  },
});
