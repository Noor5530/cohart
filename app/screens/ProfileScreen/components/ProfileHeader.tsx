import React, { useState, useEffect } from "react";
import {
  View,
  Text,
  Image,
  StyleSheet,
  TouchableOpacity,
  Linking,
} from "react-native";
import {
  widthPercentageToDP,
  heightPercentageToDP,
} from "react-native-responsive-screen";
import { iconColor, primary } from "../../../config/colors";
import { appFonts } from "../../../components/text";
import Entypo from "react-native-vector-icons/Entypo";
import Fontisto from "react-native-vector-icons/Fontisto";
import AntDesign from "react-native-vector-icons/AntDesign";
import TitleStump from "./TitleStump";
import dummyAvator from "assets/dummyAvator.png";
import { profileHeaderProps } from "../types";
import { useCallback } from "react";
import QrModel from "./QrModel";
import { useNavigation } from "@react-navigation/core";

export default function ProfileHeader(props: profileHeaderProps) {
  const {
    full_name = "",
    username = "",
    title = "",
    location = "",
    instagram = "",
    twitter = "",
    website = "",
    follower_count = 0,
    title_stamp = 1,
    cover_image = null,
    first_name = "",
    _id = "",
  } = props;

  const [userImage, setUserImage] = useState(cover_image ? cover_image : null);
  const [isVisibleQrModal, toggleQrModal] = useState(false);
  const navigation = useNavigation();
  useEffect(() => {
    setUserImage(cover_image ? cover_image : null);
    return () => {
      setUserImage(null);
    };
  }, [cover_image]);
  const openTwitter = useCallback(() => {
    Linking.openURL(`http://twitter.com/${twitter?.trim()}`);
  }, [twitter]);
  const openInstagram = useCallback(() => {
    Linking.openURL(`https://www.instagram.com/${instagram?.trim()}`);
  }, [instagram]);

  const openWeb = useCallback(() => {
    Linking.openURL(website);
  }, [website]);
  const onPressQr = () => {
    toggleQrModal(true);
  };

  function numFormatter(follower_count: any) {
    if (isNaN(follower_count)) {
      return 0;
    } else if (follower_count >= 999 && follower_count < 1000000) {
      return (follower_count / 1000).toFixed(1) + " K"; // convert to K for follower_countber from > 1000 < 1 million
    } else if (follower_count >= 1000000) {
      return (follower_count / 1000000).toFixed(1) + " M"; // convert to M for follower_countber from > 1 million
    } else {
      return follower_count;
    }
  }
  const onClickFans = () => {
    navigation.navigate("FansScreen", {
      _id: _id
    });
  };
  return (
    <View style={styles.container}>
      <View style={styles.leftContainer}>
        {full_name && full_name != "" ? (
          <Text style={styles.name}>{full_name}</Text>
        ) : null}
        {username && username != "" ? (
          <Text style={styles.text}>{"@" + username}</Text>
        ) : null}
        {title && title != "" ? <Text style={styles.text}>{title}</Text> : null}
        {location && location != "" ? (
          <Text style={styles.text}>
            <Entypo
              name="location-pin"
              size={widthPercentageToDP("3.5")}
              color={iconColor}
            />
            {location}
          </Text>
        ) : null}
        {website != "" || instagram != "" || twitter != "" ? (
          <View style={styles.socialContainer}>
            {website && website != "" ? (
              <TouchableOpacity
                style={{ paddingRight: 10 }}
                testID={"openWeb"}
                onPress={openWeb}
              >
                <Fontisto
                  size={widthPercentageToDP(5)}
                  name="world-o"
                  color={iconColor}
                />
              </TouchableOpacity>
            ) : null}
            {twitter && twitter != "" ? (
              <TouchableOpacity
                style={{ paddingRight: 10 }}
                testID={"openTwitter"}
                onPress={openTwitter}
              >
                <AntDesign
                  size={widthPercentageToDP(5)}
                  name="twitter"
                  color={iconColor}
                />
              </TouchableOpacity>
            ) : null}

            {instagram && instagram != "" ? (
              <TouchableOpacity
                style={{ paddingRight: 10 }}
                testID={"openInstagram"}
                onPress={openInstagram}
              >
                <AntDesign
                  size={widthPercentageToDP(5)}
                  name="instagram"
                  color={iconColor}
                />
              </TouchableOpacity>
            ) : null}
          </View>
        ) : null}
        <TouchableOpacity onPress={onClickFans}>
          <Text style={styles.bottomHeading} testID={"numFormat"}>
            {numFormatter(follower_count) + " " + "FANS"}
          </Text>
        </TouchableOpacity>
      </View>
      <View>
        <View style={styles.titleStumpContainer}>
          <TitleStump
            style={styles.titleStump}
            text={
              title_stamp == 1
                ? "PARTNER ORGANIZATON"
                : title_stamp == 2
                  ? "VERIFIED USER"
                  : title_stamp == 3
                    ? "EARLY ADOPTER"
                    : title_stamp == 4
                      ? " COHART-ER"
                      : title_stamp == 5
                        ? " ADVISOR"
                        : title_stamp == 6
                          ? "COCOISTA"
                          : "EARLY ADOPTER"
            }
          />
        </View>
        <View style={styles.imageContainer}>
          <Image
            progressiveRenderingEnabled={true}
            source={userImage ? { uri: userImage } : dummyAvator}
            resizeMode="cover"
            style={styles.imageContainer}
          />
          <TouchableOpacity
            testID={"onPressQr"}
            onPress={onPressQr}
            style={styles.qrCodeView}
          >
            <AntDesign size={22} name="qrcode" color={iconColor} />
          </TouchableOpacity>
        </View>
      </View>
      <QrModel
        isVisible={isVisibleQrModal}
        toggleModal={toggleQrModal}
        name={username}
        first_name={first_name}
        link={"https://web.cohart.co/" + username}
      />
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    paddingVertical: 22,
    flexDirection: "row",
    justifyContent: "space-between",
    minHeight: heightPercentageToDP(20),
  },
  leftContainer: { width: widthPercentageToDP(40) },
  name: {
    fontSize: widthPercentageToDP(5.1),
    color: iconColor,
    fontFamily: appFonts.InterRegular,
    paddingBottom: 4,
  },
  text: {
    fontSize: widthPercentageToDP(3),
    color: iconColor,
    fontFamily: appFonts.InterRegular,
    fontWeight: "300",
    paddingBottom: 4,
  },
  socialContainer: {
    paddingVertical: 10,
    flexDirection: "row",
    width: widthPercentageToDP(25),
  },
  bottomHeading: {
    fontSize: widthPercentageToDP(3),
    color: iconColor,
    fontFamily: appFonts.InterRegular,
  },
  titleStumpContainer: { top: -25, position: "absolute", right: -70 },
  titleStump: {
    width: widthPercentageToDP(40),
    height: widthPercentageToDP(40),
  },
  imageContainer: {
    width: widthPercentageToDP(25),
    height: widthPercentageToDP(25),
    borderRadius: widthPercentageToDP(13),
  },
  qrCodeView: {
    position: "absolute",
    bottom: 0,
    left: 0,
    backgroundColor: primary,
    width: 40,
    height: 40,
    zIndex: 5,
    borderRadius: 30,
    alignItems: "center",
    justifyContent: "center",
  },
  imageLoading: {
    position: "absolute",
    alignItems: "center",
    justifyContent: "center",
  },
});
