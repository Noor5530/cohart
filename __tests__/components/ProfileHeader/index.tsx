import "react-native";
import React from "react";
import renderer from "react-test-renderer";
import {Text} from "react-native"

import ProfileHeader from "../../../app/screens/FirstLogin/components/ProfileHeader";
jest.useFakeTimers();

test("renders correctly", () => {
  const tree = renderer.create(<ProfileHeader heading="heading" description="description"/>).toJSON();

  expect(tree).toMatchSnapshot();
});

test("renders correctly when use props renderDescription", () => {
    const tree = renderer.create(<ProfileHeader heading="heading" renderDescription={(<Text>description 1</Text>)} description="description"/>).toJSON();
  
    expect(tree).toMatchSnapshot();
  });
  
