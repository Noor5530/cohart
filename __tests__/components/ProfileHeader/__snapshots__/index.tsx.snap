// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#E6FF00",
        "height": undefined,
        "justifyContent": "flex-end",
        "paddingBottom": 30,
      },
      Object {
        "width": 750,
      },
    ]
  }
>
  <Image
    source={
      Object {
        "testUri": "../../../app/assets/profileHeaderIcon.png",
      }
    }
    style={
      Object {
        "height": undefined,
        "position": "absolute",
        "right": 0,
        "top": 0,
        "width": undefined,
      }
    }
  />
  <View>
    <Text
      style={
        Object {
          "color": "#000000",
          "fontFamily": "Inter-Bold",
          "fontSize": undefined,
          "fontStyle": "normal",
          "fontWeight": "bold",
          "height": 19,
          "lineHeight": 19,
          "paddingLeft": 38,
          "textTransform": "uppercase",
        }
      }
    >
      heading
    </Text>
    <Text
      style={
        Object {
          "color": "#000000",
          "fontFamily": "Inter-Regular",
          "fontSize": undefined,
          "fontStyle": "normal",
          "fontWeight": "normal",
          "paddingLeft": 38,
          "textTransform": "uppercase",
        }
      }
    >
      description
    </Text>
  </View>
</View>
`;

exports[`renders correctly when use props renderDescription 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#E6FF00",
        "height": undefined,
        "justifyContent": "flex-end",
        "paddingBottom": 30,
      },
      Object {
        "width": 750,
      },
    ]
  }
>
  <Image
    source={
      Object {
        "testUri": "../../../app/assets/profileHeaderIcon.png",
      }
    }
    style={
      Object {
        "height": undefined,
        "position": "absolute",
        "right": 0,
        "top": 0,
        "width": undefined,
      }
    }
  />
  <View>
    <Text
      style={
        Object {
          "color": "#000000",
          "fontFamily": "Inter-Bold",
          "fontSize": undefined,
          "fontStyle": "normal",
          "fontWeight": "bold",
          "height": 19,
          "lineHeight": 19,
          "paddingLeft": 38,
          "textTransform": "uppercase",
        }
      }
    >
      heading
    </Text>
    <Text>
      description 1
    </Text>
  </View>
</View>
`;
